src/server/status_server/mod.rs

```diff
@@ -641,6 +664,82 @@ where
+        let router = self.router.clone();
+        // Start to tls serve.
+        let server = builder.serve(make_service_fn(move |conn: &SslStream<AddrStream>| {
+            let cfg_controller = cfg_controller.clone();
+            let router = router.clone();
+            let x509 = conn.ssl().peer_certificate();
+            let cert_allowed_cn = cert_allowed_cn.clone();
+            async move {
+                // Create a status service.
+                Ok::<_, hyper::Error>(service_fn(move |req: Request<Body>| {
+                    let cfg_controller = cfg_controller.clone();
+                    let router = router.clone();
+                    let cert_allowed_cn = cert_allowed_cn.clone();
+                    let x509 = x509.clone();
+                    async move {
+                        let path = req.uri().path().to_owned();
+                        let method = req.method().to_owned();
+
+                        #[cfg(feature = "failpoints")]
+                        {
+                            if path.starts_with(FAIL_POINTS_REQUEST_PATH) {
+                                return handle_fail_points_request(req).await;
+                            }
+                        }
+
+                        match (method, path.as_ref()) {
+                            (Method::GET, "/metrics") => Ok(Response::new(dump().into())),
+                            (Method::GET, "/status") => Ok(Response::default()),
+                            (Method::GET, "/debug/pprof/heap") => {
+                                Self::dump_prof_to_resp(req).await
+                            }
+                            (Method::GET, "/config") => {
+                                Self::get_config(req, &cfg_controller).await
+                            }
+                            (Method::POST, "/config") => {
+                                if Self::check_cn(cert_allowed_cn, x509) {
+                                    Self::update_config(cfg_controller.clone(), req).await
+                                } else {
+                                    Ok(StatusServer::err_response(
+                                        StatusCode::FORBIDDEN,
+                                        "certificate role error",
+                                    ))
+                                }
+                            }
+                            (Method::GET, "/debug/pprof/profile") => {
+                                Self::dump_rsperf_to_resp(req).await
+                            }
+                            (Method::GET, path) if path.starts_with("/region") => {
+                                Self::dump_region_meta(req, router).await
+                            }
+                            _ => Ok(StatusServer::err_response(
+                                StatusCode::NOT_FOUND,
+                                "path not found",
+                            )),
+                        }
```

Comment:
This is almost identical to https://github.com/tikv/tikv/pull/8064/files#diff-d9e7f217b700ad74e968ee5cfdce6ab0R619-R652, could you unify them?

Reply:
Sure, I am looking for a easy solution to solve

Reply:
Done, thank for @sticnarf's suggestion.